jib-core/src/main/java/com/google/cloud/tools/jib/tar/TarExtractor.java

```diff
@@ -73,5 +89,34 @@ public static void extract(Path source, Path destination) throws IOException {
         }
       }
     }
+    preserveModificationTimes(destination, entries, enableReproducibleTimestamps);
+  }
+
+  /**
+   * Preserve modification time of files and directories in a tar file. If a directory is not an
+   * entry in the tar file and reproducible timestamps are enabled then it's modification timestamp
+   * is set to a constant value.
+   *
+   * @param destination target root for unzipping
+   * @param entries list of entries in tar file
+   * @param enableReproducibleTimestamps whether or not reproducible timestamps should be used
+   * @throws IOException when I/O error occurs
+   */
+  private static void preserveModificationTimes(
+      Path destination, List<TarArchiveEntry> entries, boolean enableReproducibleTimestamps)
+      throws IOException {
+    if (!Files.exists(destination)) {
+      return;
+    }
```

Comment:
This can never be true. We already extracted files into `destination`.

And the check we talked about was only for when `enableReproducibleTimestamps` is true. That is, if it's enabled, we should proceed only when either 1) `destination` doesn't exist; or 2) `destination` exists but it's empty. (Or, I'm fine with the contract where `destination` must not exist when `enableReproducibleTimestamps` is true, if that's simpler.) In any case, this precondition should be documented in the Javadoc of `extract()`. That said, the check should be at the beginning of `extract()` so that it fails fast.

Reply:
If you are still confused or there exists something unclear, don't hesitate to ask to clarify.